pkg/events/derive/hidden_kernel_module.go

```diff
@@ -38,34 +59,121 @@ func deriveHiddenKernelModulesArgs() deriveArgsFunction {
 		if err != nil {
 			return nil, err
 		}
 
-		if _, found := foundHiddenKernModsCache.Get(address); found {
-			return nil, nil // already reported this event - no need to report again
+		// revive:disable // revive complains about empty if block... but it's needed to avoid entering other blocks, if that flag is on.
+		if flags&HiddenModule != 0 {
+			// Continue submit an event to user
+		} else if flags&FullScan != 0 {
+			wakeupChannel <- ScanRequest{Flags: flags} // No need to send the address as we're doing a full generic scan
+			return nil, nil
+		} else if flags&NewMod != 0 {
+			startScanTime := address // The address field in this case is unused, so it's being utilized as the start scan time
+			err := newModsCheckForHidden(startScanTime, flags)
+			return nil, err
 		}
+		// revive:enable
 
-		foundHiddenKernModsCache.Add(address, struct{}{}) // so we won't report multiple times
+		var name string
+		nameBytes, err := parse.ArgVal[[]byte](&event, "name")
+		if err != nil {
+			name = ""
+			logger.Debugw("Failed extracting hidden module name") // Don't fail hard - submit it without a name!
+		} else {
+			name = string(nameBytes[:bytes.IndexByte(nameBytes[:], 0)]) // Remove the trailing terminating characters
+		}
+
+		var srcversion string
+		srcversionBytes, err := parse.ArgVal[[]byte](&event, "srcversion")
+		if err != nil {
+			srcversion = ""
+			logger.Debugw("Failed extracting hidden module srcversion") // Don't fail hard - submit it without a srcversion!
+		} else {
+			srcversion = string(srcversionBytes[:bytes.IndexByte(srcversionBytes[:], 0)]) // Remove the trailing terminating characters
+		}
 
 		addrHex := fmt.Sprintf("0x%x", address)
 		if len(addrHex) == 2 {
 			logger.Warnw("Failed converting module address to hex")
 		}
 
-		return []interface{}{addrHex, string(name)}, nil
+		foundHiddenKernModsCache.Add(address, struct{}{}) // So we won't report multiple times
+
+		return []interface{}{addrHex, name, srcversion}, nil
 	}
 }
 
-func InitFoundHiddenModulesCache() error {
+// newModsCheckForHidden monitors only new added modules (added while tracee is running),
+// and reports if they are hidden
+func newModsCheckForHidden(startScanTime uint64, flags uint32) error {
+	// Since in old kernels it is not possible to iterate on a hashmap, the job is done here (userspace).
+	// It goes over a map that is maintained in the eBPF code (on each module insertion/deletion the map is updated),
+	// and checks if each module was seen (in modules list), and when was it inserted (to address race conditions).
+	// Based on this information, it determines whether it's hidden or not.
+	// If found hidden, it sends a message in the channel, which causes the address and the flags to get passed
+	// to the lkm submitter program (eBPF), which sends it back to userspace,
+	// this time with flags that will cause it to get submitted to the user as an event,
+	err := capabilities.GetInstance().Specific(
```

Comment:
Should use the `EBPF` capabilities RING (it already contains the base line for all capabilities needed by picked events PLUS the ones needed for eBPF functions).